### ISSUE TITLE
[BUG] Race in `loadChart`

### ISSUE THREAD

**Issue by james-callahan** [mgoltzsche/khelm#36]
**Describe the bug**

When running multiple khelm renders (potentially of the same chart) at the same time, I get an error such as:
```
Running khelm 2.0.0-beta.6-e33800a (helm 3.5.4)
WARNING: using untrusted repository "https://charts.jetstack.io"
Downloading repository index of https://charts.jetstack.io
Downloading chart cert-manager v1.6.1 from repo https://charts.jetstack.io
khelm: load chart cert-manager: rename /root/.cache/helm/repository/khelm/charts.jetstack.io/charts/.tmp-cert-manager-v1.6.1-a6a70d098d77e500-437752009 /root/.cache/helm/repository/khelm/charts.jetstack.io/charts/cert-manager-v1.6.1-a6a70d098d77e500: file exists
```

I believe this is a [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) bug in [`loadChart`](https://github.com/mgoltzsche/khelm/blob/992b101b7a8a5497b5220ea397e2f1f8182714ea/pkg/helm/render.go#L39-L42) where you check the chart doesn't exist, go and download it; but don't expect some other process to have already downloaded it in the mean-time.

**khelm version:**
khelm 2.0.0-beta.6-e33800a (helm 3.5.4)

**Desktop (please complete the following information):**
 - OS: linux

**Additional context**
We're trying to do this in a CI pipeline on our gitops repository; we want the renders to run in parallel for quick CI runs.

**Comment by mgoltzsche** [comment]
@james-callahan thanks for reporting the issue. I'll look into it.

**Comment by mgoltzsche** [comment]
This happens [here](https://github.com/mgoltzsche/khelm/blob/v2.0.0-beta.6/pkg/helm/locate.go#L117) - I'll fix it.

**Comment by mgoltzsche** [comment]
Oh, actually it could also happen [here](https://github.com/mgoltzsche/khelm/blob/v2.0.0-beta.6/pkg/helm/repositories.go#L403).

**Comment by james-callahan** [comment]
Thanks for the quick turnaround and release! (https://github.com/mgoltzsche/khelm/releases/tag/v2.1.0)

I haven't checked, but did this affect khelm 1.x as well?

**Comment by mgoltzsche** [comment]
Yes, this also affects khelm 1. I wasn't going to update khelm 1 anymore. Do you still use it and need a fix there as well?

**Comment by james-callahan** [comment]
> Yes, this also affects khelm 1. I wasn't going to update khelm 1 anymore. Do you still use it and need a fix there as well?

Yeah sadly we still have a couple of helm 2 only charts :/

**Comment by mgoltzsche** [comment]
Okay, I also released the fix with khelm v1.0.1 now.

If you like the project you could give it a star :wink:.

**Comment by james-callahan** [comment]
Thankyou!
